python/feathub/processors/flink/table_builder/tests/test_flink_table_builder_sliding_window.py

```diff
@@ -581,21 +582,106 @@ def test_transform_with_value_counts(self):
 
         expected_result_df = pd.DataFrame(
             [
-                ["Alex", "2022-01-01 09:01:59", {"100.0": 2}, 2],
-                ["Alex", "2022-01-01 09:02:59", {"200.0": 2, "100.0": 1}, 3],
-                ["Alex", "2022-01-01 09:03:59", {"200.0": 2}, 2],
+                ["Alex", "2022-01-01 09:01:59.999", {"100.0": 2}, 2],
+                ["Alex", "2022-01-01 09:02:59.999", {"200.0": 2, "100.0": 1}, 3],
+                ["Alex", "2022-01-01 09:03:59.999", {"200.0": 2}, 2],
             ],
-            columns=["name", "time", "last_2_minute_cost_value_counts", "cnt"],
+            columns=["name", "window_time", "last_2_minute_cost_value_counts", "cnt"],
         )
         expected_result_df = expected_result_df.sort_values(
-            by=["name", "time"]
+            by=["name", "window_time"]
         ).reset_index(drop=True)
 
         table = self.flink_table_builder.build(features)
         result_df = (
             flink_table_to_pandas(table)
-            .sort_values(by=["name", "time"])
+            .sort_values(by=["name", "window_time"])
             .reset_index(drop=True)
         )
 
         self.assertTrue(expected_result_df.equals(result_df))
+
+    def test_sliding_window_with_millisecond_sliding_window_timestamp(self):
+        df = pd.DataFrame(
+            [
+                ["Alex", 100.0, "2022-01-01 09:01:00"],
+                ["Alex", 100.0, "2022-01-01 09:01:20"],
+                ["Alex", 200.0, "2022-01-01 09:02:00"],
+                ["Alex", 200.0, "2022-01-01 09:02:30"],
+            ],
+            columns=["name", "cost", "time"],
+        )
+
+        schema = Schema(["name", "cost", "time"], [String, Float64, String])
+        source = self._create_file_source(
+            df, schema=schema, keys=["name"], timestamp_format="%Y-%m-%d %H:%M:%S"
+        )
+
+        features = SlidingFeatureView(
+            name="features",
+            source=source,
+            features=[
+                Feature(
+                    name="cnt",
+                    dtype=Int64,
+                    transform=SlidingWindowTransform(
+                        expr="1",
+                        agg_func="COUNT",
+                        group_by_keys=["name"],
+                        window_size=timedelta(minutes=2),
+                        step_size=timedelta(minutes=1),
+                        limit=3,
+                    ),
+                ),
+                Feature(
+                    name="epoch_window_time",
+                    dtype=Int64,
+                    transform="UNIX_TIMESTAMP(sliding_window_timestamp)",
+                ),
+            ],
+            timestamp_field="sliding_window_timestamp",
```

Comment:
Can you add a unit test with SlidingWindow's `timestamp_format` different from the timestamp format of its source field?